src/components/Course/Course.vue

```diff
@@ -27,7 +27,15 @@
       <div class="course-main">
         <div class="course-top">
           <div class="course-left">
-            <div class="course-code" data-cyId="courseCode">{{ courseObj.code }}</div>
+            <div
+              class="course-code"
+              data-cyId="courseCode"
+              :style="{
+                color: compact ? '#3d3d3d' : '#858585',
```

Comment:
I wonder if you might want to use CSS classes instead? You could conditionally assign the class, and then in the CSS since we use SASS we can just make it a variable color alias. That would make things more future-proof.